Lombiq.NodeJs.Extensions/Docs/JavaScript.md

```diff
@@ -0,0 +1,84 @@
+
+Looking for something similar for .NET? Check out our [.NET Analyzers project](https://github.com/Lombiq/.NET-Analyzers).
+
+
+## Source and target paths
+
+The default paths for JS input and output files are *Assets/Scripts* and *wwwroot/js*, respectively. The existing folder structure in the input folder will be mirrored in the output, e.g. *Assets/Scripts/app/main.js* will be transformed into *wwwroot/js/app/main.js*, together with *wwwroot/js/app/main.min.js* and *wwwroot/js/app/main.js.map*.
+
+### Overriding the defaults - coming soon!
+
+Those defaults can be overridden by providing the following MSBuild properties in your project file:
+
+```xml
+<NodeJsExtensionsScriptsSourceFolder>Assets/Scripts</NodeJsExtensionsScriptsSourceFolder>
+<NodeJsExtensionsScriptsTargetFolder>wwwroot/js</NodeJsExtensionsScriptsTargetFolder>
+```
+
+
+## How to get started
+
+To use the `npm` scripts defined in this project, add any or all of the following entries to the `scripts` property in your project's *package.json*:
+
+- If you're using the default paths, then use these:
+
+  ```json
+  "scripts": {
+    "build": "npm run build:scripts",
+    "build:scripts": "npm explore nodejs-extensions -- pnpm run build:scripts",
+    "clean": "npm run clean:scripts",
+    "clean:scripts": "npm explore nodejs-extensions -- pnpm run clean:scripts",
+    "watch": "npm run watch:scripts",
+    "watch:scripts": "npm explore nodejs-extensions -- pnpm run watch:scripts",
+  }
+  ```
+
+- If you're using non-default paths (support for which is coming soon), then you will need to add the following entries (using the example paths above):
+
+  ```json
+  "scripts": {
+    "build": "npm run build:scripts",
+    "build:scripts": "npm explore nodejs-extensions -- npm run build:scripts:args --source=path/to/my/js-files --target=path/to/my/js-files",
+    "clean": "npm run clean:scripts",
+    "clean:scripts": "npm explore nodejs-extensions -- npm run clean:scripts:args --target=path/to/my/js-files",
+    "watch": "npm run watch:scripts",
+    "watch:scripts": "npm explore nodejs-extensions -- npm run watch:scripts:args --source=path/to/my/js-files --target=path/to/my/js-files",
+  }
+  ```
+
+To see the different configurations of default and non-default paths in action, please check out our dedicated [Samples](../../Lombiq.NodeJs.Extensions.Samples/Readme.md) [projects](../../Lombiq.NodeJs.Extensions.Samples.NuGet/Readme.md).
+
+### Integration with Visual Studio (Code)
+
+Visual Studio support ESLint out of the box. You can enable it by ticking the checkbox "Enable ESLint" under *Tools -> Options -> Text Editor -> JavaScript/TypeScript -> Linting -> General*.
```

Comment:
The remainder of the linked section is obsolete now.

Reply:
Previously, VS needed a separate file in a specific location for JS linting to work. Isn't that the case today?

Reply:
I've got it to work at one point, but after a full clean, it's broken again. So, yes, it's possible without extra files, I just have to make it permanent.

Reply:
We need some kind of common ESLint config since we don't want to have to maintain config overrides in N places. Due to the reusable and shared nature of all OSOCE projects it's not necessarily exactly the same here, but generally in all solutions we need a central, single config for ESLint. This might be achievable in different ways though, not just what we've done before.

Reply:
Why does an upfront build needs to happen though? Since NE is referenced by the projects it's used from, it should be at the root of the dependency tree, being built first.

Reply:
I've realized in the meantime, that the problematic part of the whole configuration setup is **only** with the VS ESLint integration, not with running ESLint from NE during the build itself :grin

So, yes, no upfront build should be necessary.

Reply:
I've got the VS ESLint support working now for JS files in the Samples project, with both locations for the _.eslintrc_ file, in a project, and inside the solution root folder.

For some reason it doesn't work yet in the NuGet samples.